substra/cli/printers.py

```diff
@@ -53,6 +53,15 @@ def print_details(self, item, field_length, expand):
                         print(f'{padding}- {v}')
             else:
                 print(f'{name}None')
+        elif isinstance(value, dict):
+            if len(value):
+                padding = ' ' * field_length
+                print(f'{name}')
+                for k, v in value.items():
+                    print(f'{padding}- {k}: {v}')
+            else:
+                name = self.name.upper().ljust(field_length)
+                print(f'{name}None')
```

Comment:
Here is the output of this code:
```
METADATA
                            - foo: bar
```

The -foo:bar should be on the same line as METADATA here, there's no reason to have an extraline.

If you look at the code on L45-46, this is achieved by adding `end=''` to the first print, thus not printing the carriage return after `name`. Then the first item of the list is not printed with padding.

Since printing list and dicts is almost the same, you could transform the dict value as a list value and then let the existing code handle the printing.

For example:

```python
if isinstance(value, dict):
    dict = [f{k}: {v}' for k, v in value.items()]
```
```